docs/xxenabledynamicagentloading.md

```diff
@@ -0,0 +1,60 @@
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# -XX:\[+|-\]EnableDynamicAgentLoading
+
+This option enables or disables the dynamic loading of agents into a running VM.
+
+
+## Syntax
+
+        -XX:[+|-]EnableDynamicAgentLoading
+
+| Setting               | Effect  | Default                                                                            |
+|-----------------------|---------|:----------------------------------------------------------------------------------:|
+| `-XX:+EnableDynamicAgentLoading` |  Enable   |   :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>                                 |
+| `-XX:-EnableDynamicAgentLoading` |  Disable  |     |
+
+
+## Explanation
+
+With the [Attach API](attachapi.md), your application can connect to a running VM and load an agent dynamically into that VM to run tasks. Some libraries misuse the Attach API to dynamically load an agent to run tasks without the approval of the application owner.
+
+You can disable the dynamic loading of the agents into a VM after startup with the `-XX:-EnableDynamicAgentLoading` option.
+
+![Start of content that applies to Java 21 (LTS) and later](cr/java21plus.png) For Java&trade; 21 and later, the following warnings are issued when the agents are loaded dynamically into a running VM after startup without specifying the `-XX:+EnableDynamicAgentLoading` option and the same agents were not loaded at startup by using the [command-line options](interface_jvmti.md#overview):
+
+```
+WARNING: A {Java,JVM TI} agent has been loaded dynamically (file:/u/bob/agent.jar)
+WARNING: If a serviceability tool is in use, please run with -XX:+EnableDynamicAgentLoading to hide this warning
+WARNING: If a serviceability tool is not in use, please run with -Djdk.instrument.traceUsage for more information
+WARNING: Dynamic loading of agents will be disallowed by default in a future release
+```
+AIX&reg; systems cannot detect whether an agent library was previously loaded at startup or not. Therefore, the warnings are issued when an agent is loaded dynamically through the Attach API (`VirtualMachine.loadAgentLibrary(agent)`) even if the agent was loaded through the command-line option (`-agentpath:/Absolute/Path/to/agentLibrary`) at startup. ![End of content that applies to Java 21 (LTS) and later](cr/java_close_lts.png)
```

Comment:
@JasonFengJ9 - What do you mean by "only issue one warning message which is not AIX-specific capability"?

Reply:
> What do you mean by "only issue one warning message which is not AIX-specific capability"?

If an agent is loaded dynamically with the same name or absolute path, OpenJ9 VM is able to compare it with previously loaded libraries and determine if it was already loaded, in case the agent library was loaded before, no more warnings are issued.
However, if the same library is loaded differently, for example, one with a platform-independent name such as `VirtualMachine.loadAgentLibrary(agent)`, and the other with an absolute file path `-agentpath:/Absolute/Path/to/agentLibrary`, this use case requires the OS API to determine if the same library was loaded, AIX doesn't have such APIs like other platforms supported by OpenJ9, hence another warnings will be issued.